libraries/AP_Common/Location.cpp

```diff
@@ -420,3 +448,16 @@ int32_t Location::diff_longitude(int32_t lon1, int32_t lon2)
     }
     return int32_t(dlon);
 }
+
+/*
+  limit lattitude to -90e7 to 90e7
```

Comment:
This is a mathematical limit so OK from that perspective. There is also an upper limit on latitude for reliable operation and as a follow on activity after this PR has been merged, SITL testing with a polar crossing is likely to uncover unwanted behaviours..

Reply:
yes, I have another set of patches as part of my "round the world" efforts. I decided not to try to get those into this base double EKF PR, and instead do it as a separate PR. It will need some discussion on approaches to the various corner cases